src/backend/distributed/shared_library_init.c

```diff
@@ -2907,7 +2919,9 @@ ApplicationNameAssignHook(const char *newval, void *extra)
 	 * as reasonably possible, which is also why we extract global pids in the
 	 * AuthHook already (extracting doesn't require catalog access).
 	 */
-	if (FinishedStartupCitusBackend)
+	if (FinishedStartupCitusBackend &&
+		(!IsExternalClientBackend() || CachedLocalNodeIdIsValid() ||
```

Comment:
Noting that since IsExternalClientBackend() covers the background workers too, this fix will protect against 
   a background worker + conf file has application_name + SIGHUP outside a transaction 
scenario as well.